backend/controllers/prisonerProfile/prisonerCellHistory.js

```diff
@@ -32,46 +32,39 @@ module.exports = ({ oauthApi, prisonApi, logError, page = 0 }) => async (req, re
       page,
       size: 10000,
     })
-    const userRoles = await oauthApi.userRoles(res.locals)
-    const canViewCellMoveButton = userRoles && userRoles.some(role => role.roleCode === 'CELL_MOVE')
+    const [userRoles, agencyData] = await Promise.all([oauthApi.userRoles(res.locals), getAgencyDetails(cells)])
+    const staff = await Promise.all(cells.map(cell => prisonApi.getStaffDetails(res.locals, cell.movementMadeBy)))
 
-    // Collect a list of the unique agencies, to remove
-    // the need to make duplicate API calls
-    const agencyData = await getAgencyDetails(cells)
+    const cellData = cells.map(cell => {
+      const staffDetails = staff.find(user => cell.movementMadeBy === user.username)
+      const agency = agencyData.find(agencyDetails => cell.agencyId === agencyDetails.agencyId)
 
-    const currentLocation = cells.find(cell => cell.assignmentEndDateTime === undefined)
+      return {
+        establishment: agency.description,
+        location: extractLocation(cell.description, cell.agencyId),
+        movedIn: cell.assignmentDateTime && formatTimestampToDateTime(cell.assignmentDateTime),
+        movedOut: cell.assignmentEndDateTime && formatTimestampToDateTime(cell.assignmentEndDateTime),
+        assignmentDateTime: moment(cell.assignmentDateTime).format('YYYY-MM-DDTHH:mm:ss'),
+        assignmentEndDateTime: cell.assignmentEndDateTime
+          ? moment(cell.assignmentEndDateTime).format('YYYY-MM-DDTHH:mm:ss')
+          : undefined,
+        livingUnitId: cell.livingUnitId,
+        agencyId: cell.agencyId,
+        movedInBy: formatName(staffDetails.firstName, staffDetails.lastName),
+      }
+    })
+
+    const currentLocation = cellData.find(cell => cell.assignmentEndDateTime === undefined)
     const occupants =
       (currentLocation && (await prisonApi.getInmatesAtLocation(res.locals, currentLocation.livingUnitId, {}))) || []
 
-    const today = moment().format('YYYY-MM-DDTHH:mm:ss')
-
-    const cellData = await Promise.all(
-      cells
-        .filter(cell => cell.assignmentEndDateTime)
-        .sort((left, right) => sortByDateTime(right.assignmentDateTime, left.assignmentDateTime))
-        .map(async cell => {
-          const staffDetails = await prisonApi.getStaffDetails(res.locals, cell.movementMadeBy)
-          const agency = agencyData.find(agencyDetails => cell.agencyId === agencyDetails.agencyId)
-
-          return {
-            establishment: agency.description,
-            location: extractLocation(cell.description, cell.agencyId),
-            movedIn: cell.assignmentDateTime && formatTimestampToDateTime(cell.assignmentDateTime),
-            movedOut: cell.assignmentEndDateTime && formatTimestampToDateTime(cell.assignmentEndDateTime),
-            assignmentDateTime: moment(cell.assignmentDateTime).format('YYYY-MM-DDTHH:mm:ss'),
-            assignmentEndDateTime: cell.assignmentEndDateTime
```

Comment:
Checking for `cell.assignmentEndDateTime` but they've already been filtered out above.